File: app/src/main/java/com/gmail/fattazzo/formula1world/ergast/json/objects/Circuit.java
package com.gmail.fattazzo.formula1world.ergast.json.objects;

import com.gmail.fattazzo.formula1world.domain.F1Circuit;

/**
 * @author fattazzo
 *         <p/>
 *         date: 06/06/17
 */
public class Circuit {
    private String circuitId;
    private String url;
    private String circuitName;
    private Location location;

    public Circuit(String circuitId, String url, String circuitName, Location location) {
        this.circuitId = circuitId;
        this.url = url;
        this.circuitName = circuitName;
        this.location = location;
    }

    F1Circuit toF1Circuit() {
        F1Circuit f1Circuit = new F1Circuit();
        f1Circuit.circuitRef = this.circuitId;
        f1Circuit.name = this.circuitName;
        f1Circuit.url = this.url;
        if (this.location != null) {
            f1Circuit.location = this.location.toF1Location();
        }
        return f1Circuit;
    }

    @Override
    public String toString() {
        return "Circuit{" +
                "circuitId='" + circuitId + '\'' +
                ", url='" + url + '\'' +
                ", circuitName='" + circuitName + '\'' +
                ", location=" + location +
                '}';
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/ergast/json/objects/Circuit.kt
/*
 * Project: total-gp-world
 * File: Circuit.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.ergast.json.objects

import com.gmail.fattazzo.formula1world.domain.F1Circuit

/**
 * @author fattazzo
 *
 *
 * date: 06/06/17
 */
class Circuit internal constructor(private val circuitId: String, private val url: String, private val circuitName: String, private val location: Location?) {

    internal fun toF1Circuit(): F1Circuit {
        val f1Circuit = F1Circuit()
        f1Circuit.circuitRef = this.circuitId
        f1Circuit.name = this.circuitName
        f1Circuit.url = this.url
        if (this.location != null) {
            f1Circuit.location = this.location.toF1Location()
        }
        return f1Circuit
    }

    override fun toString(): String {
        return "Circuit{" +
                "circuitId='" + circuitId + '\'' +
                ", url='" + url + '\'' +
                ", circuitName='" + circuitName + '\'' +
                ", location=" + location +
                '}'
    }
}
